pkg/new-ui/v1beta1/util.go

```diff
@@ -94,8 +95,31 @@ func (k *KatibUIHandler) getTrialTemplatesViewList() ([]TrialTemplatesDataView,
 			return nil, err
 		}
 
+		// Iterate over the trialTemplatesConfigMapList from all namespaces and filter out the
+		// configmaps that belong to namespaces other than kubeflow and the ones that the user has
+		// access privileges.
+		var newTrialTemplatesConfigMapList []apiv1.ConfigMap
+		for _, cmap := range trialTemplatesConfigMapList.Items {
+			if ns == "kubeflow" {
+				// Add the configmaps from kubeflow namespace (no user can have access to kubeflow ns, so we hardcode it)
+				newTrialTemplatesConfigMapList = append(newTrialTemplatesConfigMapList, cmap)
+			} else {
+				// for all other namespaces check authorization rbac
+				configmapName := cmap.ObjectMeta.Name
+				user, err := IsAuthorized(consts.ActionTypeGet, ns, apiv1.ResourceConfigMaps.String(), "", configmapName, apiv1.SchemeGroupVersion, k.katibClient.GetClient(), r)
+				if err != nil {
+					log.Printf("The user: %s is not authorized to view configMap: %s in namespace: %s \n", user, configmapName, ns)
+					return nil, err
+				} else {
+					log.Printf("The user: %s is authorized to view configMap: %s in namespace: %s", user, configmapName, ns)
+					newTrialTemplatesConfigMapList = append(newTrialTemplatesConfigMapList, cmap)
+				}
```

Comment:
@apo-ger Maybe we should make the `IsAuthorized` check before we are [getting Trial templates](https://github.com/kubeflow/katib/blob/ae941c4cc66747a7e0db363ec7c9e1f877a1b9c9/pkg/new-ui/v1beta1/util.go#L92) ?
Then, you don't need to have additional condition for `kubeflow` namespace.

Reply:
IIUC the current implementation of the UI only allows users to view the available `TrialTemplates` ([/katib/fetch_trial_templates/](https://github.com/apo-ger/katib/blob/ae941c4cc66747a7e0db363ec7c9e1f877a1b9c9/pkg/new-ui/v1beta1/backend.go#L256)), right?

Reply:
I see, makes sense @apo-ger.
> IIUC the current implementation of the UI only allows users to view the available TrialTemplates ([/katib/fetch_trial_templates/](https://github.com/apo-ger/katib/blob/ae941c4cc66747a7e0db363ec7c9e1f877a1b9c9/pkg/new-ui/v1beta1/backend.go#L256)), right?

We have API to create/delete template. For example: https://github.com/apo-ger/katib/blob/ae941c4cc66747a7e0db363ec7c9e1f877a1b9c9/pkg/new-ui/v1beta1/backend.go#L280.
The old UI has feature to CRUD Trial templates, but the new UI still don't support it, tracking here: https://github.com/kubeflow/katib/issues/1440. cc @kimwnasptd 
We might need to have discussion around this. Should we show Trial templates in `kubeflow` namespace or just avoid it.

Reply:
> We might need to have discussion around this. Should we show Trial templates in kubeflow namespace or just avoid it.

+1 on having a discussion around this. Although I think the current approach makes sense:
1. An admin adds TrialTemplate (ConfigMap) in the `kubeflow` namespace. These are *visible* from all users
    * But the backend won't allow users to edit/delete ConfigMaps in the `kubeflow` namespace. For these actions there are SubjectAccessReview checks in place. 
    * If an admin wants to allow a user to manipulate the global (`kubeflow` namespace) ConfigMaps then they just need to give them RBAC permissions
2. Users will then see/interact-with ConfigMaps in other namespaces that they have permissions for